Studio-Lovelies/GG-JointJustice-Unity
# Split build and test run into different workflows to address Unity Test Runner flakiness inside GitHub Actions

## Summary

![image](https://user-images.githubusercontent.com/1689033/133704954-d2dc6287-782c-41bf-abbb-e1f4878a7627.png)

Once this changes is merged (and, if needed, [the branch protection rules of this repository](https://github.com/Studio-Lovelies/GG-JointJustice-Unity/settings/branches) are updated to reflect the following), the commit status is only set to `failure`, if the project fails to build:
![image](https://user-images.githubusercontent.com/1689033/133705155-a0a3e6cb-8f04-4519-ab10-2d18127f1105.png)
This way, test suites are still executed in parallel on PR creation and push, but don't prevent merges.

## Before/after screenshots and/or animated gif
### Before
![image](https://user-images.githubusercontent.com/1689033/133704975-88176fb6-1207-495c-b835-fcc6b3eefd26.png)

### After
![image](https://user-images.githubusercontent.com/1689033/133704983-1040a829-1346-4d94-ac3a-eeaa6a998180.png)
![image](https://user-images.githubusercontent.com/1689033/133705297-8c06ecda-f417-4ea6-8b90-579ae5ef62cc.png)

## Testing instructions

1. Fork this repository
2. Push the changes of this branch to `develop` on your fork
3. Execute both workflows on a change with failing tests
4. Notice `Run PlayMode and EditMode tests` failing, but not preventing `Generate game builds` to continue execution

## Additional information

- `[ ]` Changes UI
- `[ ]` Introduces new feature
- `[ ]` Removes existing feature
- `[ ]` Has associated resource:
